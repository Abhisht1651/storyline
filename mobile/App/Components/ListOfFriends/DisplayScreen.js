import React from 'react'
import { View } from 'react-native'
import ListOfFriends from './ListOfFriends'
import styles from '../Styles/ListOfFriendsStyles/DisplayScreenStyle'

const DisplayScreen = () => {
  return (
    <View style={styles.container}>
      <ListOfFriends
        onPress={() => console.log('Button Pressed')}
        onCancelPress={() => console.log('Press Cancel')}
      />
    </View>
  )
}

export default DisplayScreen
