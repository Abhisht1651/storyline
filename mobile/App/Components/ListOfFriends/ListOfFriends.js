// @flow

import React, { useState, useEffect } from 'react'
import { View, Text, TouchableOpacity, Image } from 'react-native'
import styles from '../Styles/ListOfFriendsStyles/ListOfFriendsStyle'
import Images from '../../Themes/Images'

type Props = {
  image: Image,
  name: string,
  onPress: () => {},
  onCancelPress: () => {},
  btnText1: string,
  btnText2: string
}

const ListOfFriends = (props: Props) => {
  const { image, name, onPress, onCancelPress, btnText1, btnText2 } = props

  const [buttonPressed, setButtonPressed] = useState(false)
  const [buttonStyle, setButtonStyle] = useState({ borderColor: '#7B46E4' })
  const [textStyle, setTextStyle] = useState({ color: '#7B46E4' })
  const [buttonText, setButtonText] = useState(btnText1)

  useEffect(() => {
    if (buttonPressed) {
      setButtonStyle({ borderColor: '#DCDCE9' })
      setTextStyle({ color: '#8F8FAF' })
      setButtonText(btnText2)
    } else {
      setButtonStyle({ borderColor: '#7B46E4' })
      setTextStyle({ color: '#7B46E4' })
      setButtonText(btnText1)
    }
  }, [buttonPressed])

  const onPressHandler = () => {
    setButtonPressed(!buttonPressed)
    if (buttonPressed) onCancelPress()
    else if (!buttonPressed) onPress()
  }
  return (
    <View style={styles.container}>
      <View style={styles.photo}>
        <Image style={styles.image} source={image} />
      </View>
      <View style={styles.name}>
        <Text style={styles.nameText}>{name}</Text>
      </View>
      <TouchableOpacity
        style={[styles.button, buttonStyle]}
        activeOpacity={1}
        onPress={onPressHandler}>
        <Text style={[styles.buttonText, textStyle]}>{buttonText}</Text>
      </TouchableOpacity>
    </View>
  )
}

ListOfFriends.defaultProps = {
  image: Images.picture,
  name: 'Name',
  btnText1: 'Add',
  btnText2: 'Requested'
}

export default ListOfFriends
