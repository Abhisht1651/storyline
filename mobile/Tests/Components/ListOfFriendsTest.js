import { shallow } from 'enzyme'
import React from 'react'

import { TouchableOpacity } from 'react-native'
import ListOfFriends from '../../App/Components/ListOfFriends/ListOfFriends'

describe('ListOfFriends', () => {
  it('should match to snapshot', () => {
    const component = shallow(<ListOfFriends />)
    expect(component).toMatchSnapshot()
  })

  it('should trigger function onPress', () => {
    const onPressed = jest.fn()
    const wrapper = shallow(<ListOfFriends onPress={onPressed} />)
    wrapper.find(TouchableOpacity).props().onPress()
    expect(onPressed.mock.calls.length).toBe(1)
  })
})
