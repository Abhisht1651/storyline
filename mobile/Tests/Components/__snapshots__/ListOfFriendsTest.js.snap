// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListOfFriends should match to snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "borderColor": "#DCDCE9",
      "borderRadius": 11.5,
      "borderWidth": 1,
      "height": 264.5,
      "width": 172.5,
    }
  }
>
  <View
    style={
      Object {
        "aspectRatio": 1,
        "height": 98.57142857142857,
        "marginBottom": 13.142857142857142,
        "marginTop": 16.428571428571427,
      }
    }
  >
    <Image
      source={
        Object {
          "default": "",
        }
      }
      style={
        Object {
          "aspectRatio": 1,
          "borderRadius": 60,
          "height": 98.57142857142857,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "height": 50.92857142857142,
        "marginBottom": 14.785714285714285,
        "width": "62.28%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#1F1F60",
          "fontFamily": "BananaGrotesk-Light",
          "fontSize": 23,
          "textAlign": "center",
        }
      }
    >
      Name
    </Text>
  </View>
  <ForwardRef
    activeOpacity={1}
    onPress={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 16,
          "borderWidth": 1,
          "height": 49.285714285714285,
          "justifyContent": "center",
          "marginBottom": 21.357142857142858,
          "maxWidth": "95%",
          "minWidth": "71%",
          "paddingLeft": 8.214285714285714,
          "paddingRight": 8.214285714285714,
        },
        Object {
          "borderColor": "#7B46E4",
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#7B46E4",
            "fontFamily": "BananaGrotesk-Bold",
            "fontSize": 23,
            "letterSpacing": -0.08,
          },
          Object {
            "color": "#7B46E4",
          },
        ]
      }
    >
      Add
    </Text>
  </ForwardRef>
</View>
`;
